src/routes.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import Dashboard from './containers/Dashboard'
+import {Login} from './containers/Login'
+import Profile from './containers/Profile'
+import { AuthStore } from './repos/UserRepo'
+import {
+Route,
+Switch,
+Redirect 
+} from 'react-router-dom'
+
+
+const  renderIfAuthenticated = (Component) => props => AuthStore.isAuthenticated ? 
+(<Component {...props} />) : 
+(<Redirect
+    to={{
+      pathname: "/login",
+      state: { from: props.location }
+    }}
+  />
+)
+
+const PrivateRoute = ({ component: Component, ...rest }) => (
```

Comment:
based on the official router example: https://reacttraining.com/react-router/web/example/auth-workflow